src/AspNetCoreModuleV2/AspNetCore/Inc/applicationinfo.h

```diff
@@ -145,16 +132,18 @@ class APPLICATION_INFO
     HRESULT FindRequestHandlerAssembly(STRU& location);
     HRESULT FindNativeAssemblyFromGlobalLocation(PCWSTR libraryName, STRU* location);
     HRESULT FindNativeAssemblyFromHostfxr(HOSTFXR_OPTIONS* hostfxrOptions, PCWSTR libraryName, STRU* location);
+    BOOL    LoadAppOffline(LPWSTR strFilePath);
 
     static VOID DoRecycleApplication(LPVOID lpParam);
 
     mutable LONG            m_cRefs;
     STRU                    m_struInfoKey;
+    STRA                    m_strAppOfflineContent;
     BOOL                    m_fAppOfflineFound;
+    BOOL                    m_fAppOfflineLoaded;
```

Comment:
Is m_fAppOfflineLoaded set to false after app_offline is removed?

Reply:
once appoffline is detected, the instance of applicationinfo will be recycled which may take time. no need to reset m_fAppOfflineLoaded as we don't want bad thing happen in the middle